pkg/storage/crdb.go

```diff
@@ -16,17 +16,23 @@ import (
 	"github.com/cockroachdb/cockroach-go/crdb"
 )
 
-func createJobTx(tx *sql.Tx, cidBytes []byte, pub Pub, timestamp *int64) error {
+func createJobTx(tx *sql.Tx, cidBytes []byte, pub Pub, timestamp *int64, fname string, cacheDuration int64) error {
 	row := tx.QueryRow(
 		"SELECT id FROM namespaces WHERE name = $1", pub.Namespace)
 	var nsID int
 	if err := row.Scan(&nsID); err != nil {
 		return errors.Wrap(err, "error while querying namespace")
 	}
 
+	expiresAt, cachePath := &sql.NullTime{}, &sql.NullString{}
+	if cacheDuration > 0 {
```

Comment:
if file should be cached we'll store both the path and calculate the expiration time